src/Domain/CoordinateMaps/TimeDependent/ShapeMapTransitionFunctions/SphereTransition.cpp

```diff
@@ -75,24 +74,46 @@ std::array<DataVector, 3> SphereTransition::gradient(
 template <typename T>
 T SphereTransition::call_impl(const std::array<T, 3>& source_coords) const {
   const T mag = magnitude(source_coords);
-  check_magnitudes(mag);
-  return a_ + b_ / mag;
+  T result = a_ + b_ / mag;
+  for (size_t i = 0; i < get_size(mag); ++i) {
+    if (get_element(mag, i) + eps_ < r_min_) {
+      get_element(result, i) = 1.0;
+    } else if (get_element(mag, i) - eps_ > r_max_) {
+      get_element(result, i) = 0.0;
+    }
+  }
+  return result;
 }
 
 template <typename T>
 T SphereTransition::map_over_radius_impl(
     const std::array<T, 3>& source_coords) const {
   const T mag = magnitude(source_coords);
-  check_magnitudes(mag);
-  return a_ / mag + b_ / square(mag);
+  T result = a_ / mag + b_ / square(mag);
+  for (size_t i = 0; i < get_size(mag); ++i) {
+    if (get_element(mag, i) + eps_ < r_min_) {
+      get_element(result, i) = 1.0 / get_element(mag, i);
+    } else if (get_element(mag, i) - eps_ > r_max_) {
+      get_element(result, i) = 0.0;
+    }
+  }
+  return result;
 }
 
 template <typename T>
 std::array<T, 3> SphereTransition::gradient_impl(
     const std::array<T, 3>& source_coords) const {
   const T mag = magnitude(source_coords);
-  check_magnitudes(mag);
-  return -b_ * source_coords / cube(mag);
+  std::array<T, 3> result = -b_ * source_coords / cube(mag);
+  for (size_t i = 0; i < get_size(mag); ++i) {
+    if (get_element(mag, i) + eps_ < r_min_ or
+        get_element(mag, i) - eps_ > r_max_) {
```

Comment:
this won't work if r_max_ is greater than 100.0 You are using an absolute tolerance, and so 100-eps_ is identically 100. Same issue with `r_min_` if it's big enough.. You can do something like `r_max_ * (1.0 + eps_)` to make it a relative scale, then it's safe. You can then probably also tighten `eps` to like `2*epsilon()`. Having said all that, this still will break smoothness inside of a block, which is an issue.